### PR TITLE
[tagger] add caching to entityTags to avoid joining arrays for every lookup

### PR DESCRIPTION
**Pull request by xvello**
### What does this PR do?

Store the result of tag concatenation from several sources inside the `entityTags` object for reuse in the next lookup. Cache is invalidated in write, and re-computed on the following read.

As we are storing `[]string`s, memory cost is pretty small (references). `cachedLow` is a slice of `cachedAll` (which stores low card tags + high card tags).
### Motivation

Optimise dsd for higher throughputs

### CODE REVIEW

File: pkg/tagger/tagstore_test.go [hunk 1]
```diff
@@ -59,10 +59,8 @@ func (s *StoreTestSuite) TestLookup() {
 		LowCardTags: []string{"tag"},
 	})
 
-	tagsHigh, sourcesHigh, err := s.store.lookup("test", true)
-	assert.Nil(s.T(), err)
-	tagsLow, sourcesLow, err := s.store.lookup("test", false)
-	assert.Nil(s.T(), err)
+	tagsHigh, sourcesHigh := s.store.lookup("test", true)
+	tagsLow, sourcesLow := s.store.lookup("test", false)
```
Comment (mfpierre) [inline pkg/tagger/tagstore_test.go]: Could you add a simple test case about cache hit/miss state in the unit test?

**Review by mfpierre — CHANGES_REQUESTED** [review]
Looks good overall  but having some tests on the cache hit/miss would be great

File: pkg/tagger/tagstore_test.go (hunk 1 above)
Comment (xvello) [reply pkg/tagger/tagstore_test.go]: done, thanks

**Review by mfpierre — APPROVED** [review]
🗄

**Pull request merged**